spec/models/job_run_spec.rb

```diff
@@ -1,58 +1,49 @@
+  end
 
-  context "validation" do
-    let(:user) do
-      User.new(sunet_id: "Jdoe")
-    end
+  it { is_expected.to belong_to(:bundle_context) }
 
-    let(:bc) do
-      BundleContext.new(project_name: "SmokeTest",
-                        content_structure: 1,
-                        bundle_dir: "spec/test_data/images_jp2_tif",
-                        staging_style_symlink: false,
-                        content_metadata_creation: 1,
-                        user: user)
-    end
 
-    let(:discovery_report_run) do
-      JobRun.new(output_location: "/path/to/report",
-                 bundle_context: bc,
-                 job_type: "discovery_report")
+  describe 'enqueue!' do
+    it 'does nothing if unpersisted' do
+      expect(DiscoveryReportJob).not_to receive(:perform_later)
+      job_run.enqueue!
     end
-
-    it "is not valid if it doesn't have the required fields" do
-      expect(JobRun.new).not_to be_valid
-      expect(discovery_report_run).to be_valid
+    it 'calls the correct job for job_type' do
+      allow(job_run).to receive(:persisted?).and_return(true)
+      expect(DiscoveryReportJob).to receive(:perform_later).with(job_run)
+      job_run.enqueue!
+      job_run.job_type = 'preassembly'
+      expect(PreassemblyJob).to receive(:perform_later).with(job_run)
+      job_run.enqueue!
     end
+  end
 
-    context "#job_type enum" do
-      it "defines expected values" do
-        is_expected.to define_enum_for(:job_type).with(
-          "discovery_report" => 0,
-          "preassembly" => 1
-        )
-      end
+  describe '#job_type enum' do
+    it 'defines expected values' do
+      is_expected.to define_enum_for(:job_type).with(
+        'discovery_report' => 0,
+        'preassembly' => 1
+      )
     end
+  end
 
-    it { is_expected.to belong_to(:bundle_context) }
-
-    context 'job_type' do
-      it 'valid values' do
-        expect(JobRun.new(job_type: 0, bundle_context: bc)).to be_valid
-        expect(JobRun.new(job_type: 1, bundle_context: bc)).to be_valid
-        expect(JobRun.new(job_type: 'discovery_report', bundle_context: bc)).to be_valid
-        expect(JobRun.new(job_type: 'preassembly', bundle_context: bc)).to be_valid
-      end
-      it 'throws ArgumentError if value missing from enum' do
-        expect { JobRun.new(job_type: 3, bundle_context: bc) }.to raise_error(ArgumentError, /'3' is not a valid job_type/)
-        expect { JobRun.new(job_type: 'foo', bundle_context: bc) }.to raise_error(ArgumentError, /'foo' is not a valid job_type/)
-      end
-      it 'is required' do
-        expect(JobRun.new(bundle_context: bc)).not_to be_valid
-      end
+  context 'validation' do
+    it 'is not valid without all required fields' do
+      expect(described_class.new).not_to be_valid
+      expect(described_class.new(bundle_context: bc)).not_to be_valid
+      expect(job_run).to be_valid
```

Comment:
totally fine with seeing this in a follow-on, but it'd be nice to show (by way of specs) that `output_location` is not required for validity.